semantics/src/main/scala/aqua/semantics/expr/func/PushToStreamSem.scala

```diff
@@ -33,7 +33,10 @@ class PushToStreamSem[S[_]](val expr: PushToStreamExpr[S]) extends AnyVal {
           streamToken,
           StreamType(element match {
             case StreamType(e) => ArrayType(e)
```

Comment:
is this case tested? Looks like it shouldn't be like this. As I understand, if we want to push a stream to a stream, it will be flattened to an array. It looks like it will be better to have manual casting `stream.toArray`. But I'm not sure if it should be done like this

Reply:
This case is for error reporting. We checked that `stream` is not `StreamType` and now call `ensureTypeMatches` on it and some `StreamType`. This will always fail. This `match` is just to construct reasonable `StreamType` to match against, to report reasonable error.